common/src/main/java/org/opensearch/ml/common/agent/MLAgent.java

```diff
@@ -190,67 +228,61 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (appType != null) {
             builder.field(APP_TYPE_FIELD, appType);
         }
+        // is_hidden field isn't going to be set by user. It will be set by the code.
+        if (isHidden != null) {
+            builder.field(MLModel.IS_HIDDEN_FIELD, isHidden);
+        }
         builder.endObject();
         return builder;
     }
 
     public static MLAgent parse(XContentParser parser) throws IOException {
-        String name = null;
-        String type = null;
-        String description = null;;
+        return parseCommonFields(parser, true); // true to parse isHidden field
+    }
+
+    public static MLAgent parseFromUserInput(XContentParser parser) throws IOException {
+        return parseCommonFields(parser, false); // false to skip isHidden field
+    }
+
+    private static MLAgent parseCommonFields(XContentParser parser, boolean parseHidden) throws IOException {
+        String name = null, type = null, description = null;
         LLMSpec llm = null;
         List<MLToolSpec> tools = null;
         Map<String, String> parameters = null;
         MLMemorySpec memory = null;
-        Instant createdTime = null;
-        Instant lastUpdateTime = null;
+        Instant createdTime = null, lastUpdateTime = null;
         String appType = null;
+        boolean isHidden = false;
 
         ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
         while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
             String fieldName = parser.currentName();
             parser.nextToken();
 
             switch (fieldName) {
-                case AGENT_NAME_FIELD:
-                    name = parser.text();
-                    break;
-                case AGENT_TYPE_FIELD:
-                    type = parser.text();
-                    break;
-                case DESCRIPTION_FIELD:
-                    description = parser.text();
-                    break;
-                case LLM_FIELD:
-                    llm = LLMSpec.parse(parser);
-                    break;
+                case AGENT_NAME_FIELD: name = parser.text(); break;
```

Comment:
Put multiple lines into one line looks reduce line numbers but also makes it hard to read, generally we use separate line.